reporting-bugs/readme.md

```diff
@@ -0,0 +1,263 @@
+
+> I was on the site, and then I clicked something, and it didn't work!
+
+<!--{{</note>}}-->
+
+### Why are these not useful?
+
+<!--{{<note type="activity" title="Inversion 2, 10m" >}}-->
+
+1.  Set a [timer for 10 minutes](https://www.google.com/search?q=timer+for+10+minutes).
+2.  Popcorn around the room and each person give a reason the previous question was bad until the timer goes off. For example:
+
+> What site are you on? What are you trying to do? How can I read your post and re-create the circumstances. Be specific! Share what you see.
+
+ <!--{{</note>}}-->
+
+### Expected behavior
+
+_What you expected to happen_
+
+When you're writing a bug report, you need to be able to explain what you expected to happen after you executed the steps you just described. You might feel this is so obvious that you don't need to write it down, but this assumption is drastically, drastically wrong.
+
+Here's an example of expected behavior:
+
+> I expected to be able to log in with my GitHub account and be redirected to the logged-in view of my dashboard on my site.
+
+### Actual behavior
+
+_What actually happened_
+
+Here's an example of actual behavior:
+
+> Instead, I was redirected back to the login page with an error message that says `To confirm your identity, sign in with the same account you used originally.`
+
+<!--{{<note type="activity" title="Spot the difference, 5m" >}}-->
+
+Look at the difference between the two examples above. What do you notice? _Apart from the user's reported problem_, what is the difference between the actual behaviour and the expected behaviour? It's subtle.
+
+1. Set a [timer for 3 minutes](https://www.google.com/search?q=timer+for+3+minutes).
+1. Popcorn around the room until you spot the difference.
+
+<!--{{</note>}}-->
+
+### Error messages
+
+_What the computer told you_
+
+Have you ever considered, and this is just a wild idea, reading the error messages that your computer gives when it goes wrong?
+
+Just kidding! We know all trainees are implacably opposed to reading error messages. But here's the thing:
+
+1. When you're writing a bug report, you need to include the error messages you're seeing. This is crucial information for anyone trying to help you.
+1. You also need to **read the message yourself** first. If you cannot understand it, you need to say _what_ you don't understand about it. This is the gap in your understanding that you need help with.
+
+<details>
+<summary>Here's a common exchange we see:</summary>
+
+#### Trainee
+
+I am unable to run my code. Every time it's saying this; npm ERR!
+
+```
+Missing script: "dev"
+npm ERR!
+npm ERR! To see a list of scripts, run:
+npm ERR!   npm run
+npm ERR! A complete log of this run can be found in:
+npm ERR!     /Users/student/.npm/_logs/2024-02-26T19_28_05_701Z-debug-0.log
+```
+
+#### Mentor
+
+What happened when you followed the instructions in the error message?
```

Comment:
My only hesitation here is that npm in particular often has some actively unhelpful suggestions in their messages. For example, when you install something with a vulnerability it tells you to `npm audit fix` (which by definition cannot help if you've only just done the install; the latest versions are already installed) or even `npm audit fix --force` (which can outright break stuff\*)
I think it needs to be:

- Read the error message
- _Identify_ suggestions
- Find out what they would do (this also means being able to form a hypothesis to test)
- Then (maybe) do them

---

\* _right now if you start a new CRA 5.0.1 app it tells you you have `8 vulnerabilities (2 moderate, 6 high)` and suggests you `npm audit fix --force` which **down**grades to `react-scripts@3.0.1` and now you have `132 vulnerabilities (1 low, 104 moderate, 23 high, 4 critical)` 🤷_